src/sarif-converter.ts

```diff
@@ -0,0 +1,376 @@
+import { DictionaryStringTo } from './dictionary-types';
+import { ScannerOptions } from './exposed-apis';
+import {
+    AxeCoreRuleResult,
+    AxeNodeResult,
+    FormattedCheckResult,
+    ScannerResults,
+} from './ruleresults';
+import * as CustomSarif from './sarif/custom-sarif-types';
+import { SarifLog } from './sarif/sarifLog';
+import * as Sarif from './sarif/sarifv2';
+import { StringUtils } from './string-utils';
+import { WCAG, WCAGData } from './wcag';
+
+export class SarifConverter {
+    private wcagList: WCAGData;
+
+    constructor(wcagList: WCAGData) {
+        this.wcagList = wcagList;
+    }
+
+    public convert(results: ScannerResults, options: ScannerOptions): SarifLog {
+        const log: SarifLog = {
+            version: CustomSarif.SarifLogVersion.v2,
+            runs: [this._convertRun(results, options)],
+        };
+        return log;
+    }
+
+    private _convertRun(
+        results: ScannerResults,
+        options: ScannerOptions,
+    ): Sarif.Run {
+        const files: DictionaryStringTo<Sarif.File> = {};
+        files[results.targetPageUrl] = {
+            mimeType: 'text/html',
+            properties: {
+                tags: ['target'],
+                title: results.targetPageTitle,
+            },
+        };
+
+        let properties: DictionaryStringTo<string> = {};
+
+        if (options && options.scanName !== undefined) {
+            properties = {
+                scanName: options.scanName,
+            };
+        }
+
+        const run: Sarif.Run = {
+            tool: {
+                name: 'Scanner',
+                fullName: 'Scanner',
+                semanticVersion: '1.0.0',
+                version: '1.0.0',
+                properties: {
+                    downloadUri: 'https://accessibilityinsights.io',
+                },
```

Comment:
I think it would be more appropriate to leave accessibilityinsights.io out of this project. I'd recommend:

```typescript
name: 'axe',
fullName: 'axe-core',
semanticVersion: '3.2.2',
version: `3.2.2',
properties: {
    downloadUri: '[https://www.deque.com/axe/](https://www.deque.com/axe/)',
},
```

Reply:
That was just for first commit though.